cmd/contour/shutdownmanager.go

```diff
@@ -66,21 +70,38 @@ func newShutdownManagerContext() *shutdownmanagerContext {
 	}
 }
 
-// handles the /healthz endpoint which is used for the shutdown-manager's liveness probe
+// healthzHandler handles the /healthz endpoint which is used for the shutdown-manager's liveness probe.
 func (s *shutdownmanagerContext) healthzHandler(w http.ResponseWriter, r *http.Request) {
 	http.StatusText(http.StatusOK)
 	if _, err := w.Write([]byte("OK")); err != nil {
 		s.Error(err)
 	}
 }
 
-// shutdownHandler handles the /shutdown endpoint which should be called from a pod preStop hook,
-// where it will block pod shutdown until envoy is able to drain connections to below the min-open threshold
-func (s *shutdownmanagerContext) shutdownHandler(w http.ResponseWriter, r *http.Request) {
+// shutdownReadyHandler handles the /shutdown endpoint which is used by Envoy to determine if it can terminate.
+// Once enough connections have drained based upon configuration, a file will be written to "/ok" in
+// the shutdown manager's file system. Any HTTP request to /shutdown will use the existence of this
+// file to understand if it is safe to terminate.
```

Comment:
It's worth mentioning that we're using a file here because the pre-stop exec hook runs in a different process than the `/shutdown` handler so we need a means to communicate between them.

Reply:
Updated the comment.